examples/ensemble/plot_hgbt_regression.py

```diff
@@ -0,0 +1,401 @@
+plt.plot(-hgbt.validation_score_)
+plt.xlabel("number of iterations")
+plt.ylabel("root mean squared error")
+_ = plt.title(f"Loss of hgbt with early stopping (n_iter={hgbt.n_iter_})")
+
+# %%
+# We can then overwrite the value for `max_iter` to a reasonable value and avoid
+# the extra computational cost of the inner validation. In this case, rounding
+# up the number of iterations to 400 may account for variability of the training
+# set:
+
+common_params["max_iter"] = 400
+common_params["early_stopping"] = False
+hgbt = HistGradientBoostingRegressor(**common_params)
+
+# %%
+# .. note:: The inner validation done during early stopping is not optimal for
+#    time series with the implementation as of scikit-learn v1.3.
+#
+# Support for missing values
+# ==========================
+# HGBT models have native support of missing values. During training, the tree
+# grower decides where samples with missing values should go (left or right
+# child) at each split, based on the potential gain. When predicting, these
+# samples are sent to either child accordingly. If a feature had no missing
+# values during training, samples with missing values for that feature are sent
+# to the child with the most samples.
+#
+# The present example shows how HGBT regressions deal with values missing
+# completely at random (MCAR), i.e. the missingness does not depend on the
+# observed data or the unobserved data. We can simulate such scenario by
+# randomly replacing values from randomly selected features with `Nan` values.
+
+import numpy as np
+
+from sklearn.metrics import root_mean_squared_error
+
+rng = np.random.RandomState(42)
+first_week = slice(0, 336)  # first week in the test set as 7 * 48 = 336
+missing_fraction_list = [0, 0.02, 0.05]
+
+
+def generate_missing_values(X, missing_fraction):
+    total_cells = X.shape[0] * X.shape[1]
+    num_missing_cells = int(total_cells * missing_fraction)
+    row_indices = rng.choice(X.shape[0], num_missing_cells, replace=True)
+    col_indices = rng.choice(X.shape[1], num_missing_cells, replace=True)
+    X_missing = X.copy()
+    X_missing.iloc[row_indices, col_indices] = np.nan
+    return X_missing
+
+
+fig, ax = plt.subplots(figsize=(12, 6))
+ax.plot(y_test.values[first_week], label="Actual transfer")
+
+for missing_fraction in missing_fraction_list:
+    X_missing = generate_missing_values(X_train, missing_fraction)
+    hgbt.fit(X_missing, y_train)
+    y_pred = hgbt.predict(X_test[first_week])
+    rmse = root_mean_squared_error(y_test[first_week], y_pred)
+    ax.plot(
+        y_pred[first_week],
+        label=f"missing_fraction={missing_fraction}, RMSE={rmse:.2f}",
+        alpha=0.5,
+    )
+ax.set(
+    title="Daily energy transfer predictions on data with MCAR values",
+    xticks=[(i + 0.2) * 48 for i in range(7)],
+    xticklabels=["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"],
+    xlabel="Time of the week",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend(loc="lower right")
```

Comment:
We need to have a small conclusion for this section.

Reply:
Addressed in [11d165c](https://github.com/scikit-learn/scikit-learn/pull/26991/commits/11d165c614a5fd34b8854c91e4cc94f5de44fe3e).